auth/ldap/ldap.go

```diff
@@ -219,9 +219,16 @@ func (lm *Manager) connect() (*ldap.Conn, error) {
 
 	// switch to TLS if specified; this needs to have certs in place
 	if lm.Config.StartTLS {
-		log.Info("Upgrading to TLS mode")
+
 		// NOTE: InsecureSkipVerify should be used only for testing
-		err = ldapConn.StartTLS(&tls.Config{InsecureSkipVerify: lm.Config.InsecureSkipVerify})
+		if common.IsEmpty(lm.Config.TLSCertIssuedTo) {
+			log.Warn("Upgrading to TLS mode with `InsecureSkipVerify=true`")
+			err = ldapConn.StartTLS(&tls.Config{InsecureSkipVerify: true})
```

Comment:
As discussed, I really *really* think enabling `InsecureSkipVerify` should be explicitly controlled by the end user (probably by a boolean on the config page).  Making that change because they omitted a seemingly-unrelated field is kind of a surprising (in a bad way) UX.